scripts/generate-style-code.js

```diff
@@ -6,12 +6,14 @@ const spec = require('mapbox-gl-style-spec').latest;
 var parseCSSColor = require('csscolorparser').parseCSSColor;
 
 global.camelize = function (str) {
+  str = str === undefined ? 'undefined' : str;
```

Comment:
This was to have `Undefined` be the value for `text-pitch-alignment` when it's not defined in the style and be able to detect when inheritance should kick in (per https://github.com/mapbox/mapbox-gl-style-spec/pull/460#issuecomment-225238097).

It's definitely the main eyebrow-raiser in this PR :smile: would love to know if there's a better way to achieve this.